servicex/resources/users/accept_user.py

```diff
@@ -1,32 +1,21 @@
-import traceback
-
-import sys
-from flask_jwt_extended import (jwt_optional)
 from flask_restful import reqparse
 from sqlalchemy.orm.exc import NoResultFound
 
-from servicex.models import UserModel
 from servicex.resources.servicex_resource import ServiceXResource
+from servicex.decorators import admin_required
+from servicex.models import UserModel
 
 parser = reqparse.RequestParser()
 parser.add_argument('email', help='This field cannot be blank', required=True)
 
 
 class AcceptUser(ServiceXResource):
-    @jwt_optional
+    @admin_required
     def post(self):
-        is_auth, auth_reject_message = self._validate_user_is_admin()
-        if not is_auth:
-            return {'message': f'Authentication Failed: {str(auth_reject_message)}'}, 401
+        data = parser.parse_args()
+        email = data['email']
         try:
-            data = parser.parse_args()
```

Comment:
Shouldn't we try to catch parser errors and report them?

Reply:
Currently, if there is a parse error (e.g. no email is included in the POST data), reqparse throws `werkzeug.exceptions.BadRequest`. This is caught by the except block, which returns the following:
`<Response 109 bytes [500 INTERNAL SERVER ERROR]> b'{"message": "400 Bad Request: The browser (or proxy) sent a request that this server could not understand."}\n'`
This is because our except block takes precedence over reqparse's built-in error handling. If the final except block is removed, the same request results in:
`<Response 53 bytes [400 BAD REQUEST]> b'{"message": {"email": "This field cannot be blank"}}\n'`
This happens regardless of whether the call to `parse_args()` is made in the try block or outside of it, since we no longer catch it ourselves.

Reply:
Of course, without the except block, other errors are no longer caught properly. I think the best solution is to move the parser code out of the try block so that we aren't catching the BadRequest, but we're still catching other errors.